### PR TITLE
[WIP] fixes #16854 - sets up atomic registry

### PR DESCRIPTION
**Pull request by thomasmckay**
- [ ] Need to set up registry security: http://docs.projectatomic.io/registry/latest/registry_quickstart/administrators/system_configuration.html#registry or https://docs.openshift.com/container-platform/3.3/install_config/registry/securing_and_exposing_registry.html
- [ ] Related to above, serve self-signed certs for docker clients
- [ ] Modify /etc/sysconfig/docker  ADD_REGISTRY='--add-registry registry.example.com:5001 --add-registry registry.example.com:5000'

### CODE REVIEW

File: playbooks/roles/atomic_registry/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+- name: 'enable Aatomic Registry services'
+  service: name=atomic-registry-master.service enabled=yes state=started
+
+- name: 'delay 15 seconds moment'
+  shell: sleep 15
+
+# https://github.com/openshift/origin/issues/11542
+- name: 'handle non-dns kubernetes service host name resolution'
+  shell: sed -i s/KUBERNETES_SERVICE_HOST=.*/KUBERNETES_SERVICE_HOST={{ ansible_default_ipv4.address }}/g /etc/sysconfig/atomic-registry-console executable=/usr/bin/bash
```
Comment (thomasmckay) [inline playbooks/roles/atomic_registry/tasks/main.yml]: todo: change to _lineinfile_

File: playbooks/katello_atomic_registry.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- hosts: all
```
Comment (thomasmckay) [inline playbooks/katello_atomic_registry.yml]: todo: add docs? what is the preferred form of documenting how to use and what variables there are?

File: playbooks/roles/atomic_registry/tasks/main.yml (hunk 1 above)
Comment (thomasmckay) [inline playbooks/roles/atomic_registry/tasks/main.yml]: done

File: playbooks/katello_atomic_registry.yml (hunk 2 above)
Comment (thomasmckay) [inline playbooks/katello_atomic_registry.yml]: will do in separate PR to update docs

Comment (thomasmckay) [inline playbooks/katello_atomic_registry.yml]: done

File: playbooks/roles/atomic_registry/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    REGISTRYPORT: "{{ atomic_registry_registry_port }}"
+    CONSOLEIMAGE: "{{ atomic_registry_console_image }}"
+    CONSOLETAG: "{{ atomic_registry_console_tag }}"
+    CONSOLEPORT: "{{ atomic_registry_console_port }}"
+    MASTERIMAGE: "{{ atomic_registry_master_image }}"
+    MASTERTAG: "{{ atomic_registry_master_tag }}"
+    MASTERPORT: "{{ atomic_registry_master_port }}"
+
+- name: 'enable Aatomic Registry services'
+  service: name=atomic-registry-master.service enabled=yes state=started
```
Comment (thomasmckay) [inline playbooks/roles/atomic_registry/tasks/main.yml]: done

File: playbooks/roles/atomic_registry/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+- name: 'delay 15 seconds moment'
+  shell: sleep 15
```
Comment (thomasmckay) [inline playbooks/roles/atomic_registry/tasks/main.yml]: It can take a short while for all three docker containers to start; they must be running before continuing. In addition I have encountered a case where one of the containers would start and then quickly die in which case the first check would succeed when it should really fail; that is the reason for the initial 15sec pause before the checking starts.

**Comment by parthaa** [comment]
I ran the thing a couple of times. Worked well with my setup.  ACK

**Comment by thomasmckay** [comment]
Need to set up security: http://docs.projectatomic.io/registry/latest/registry_quickstart/administrators/system_configuration.html#registry

**Comment by ehelms** [comment]
@thomasmckay ping -- Is this still a WIP? Do you want it to be merged?

**Comment by ekohl** [comment]
Closing for inactivity
